### PR TITLE
Fix method isDefault

### PR DESCRIPTION
**Pull request by kema91**
- Fix method isDefault
- Add some tests to check the default values correctly
- Delete the unneeded defaults from the retest.xml

With the fix it should be possible that no default values are written to the retest.xml

### CODE REVIEW

File: src/test/java/de/retest/web/DefaultValuesProviderTest.java [hunk 1]
```diff
@@ -45,4 +45,18 @@ void getDefaultValue_should_prefer_specific_over_general_values() {
 		assertThat( cut.getDefaultValue( "body", "margin-top" ) ).isEqualTo( "8px" );
 		assertThat( cut.getDefaultValue( "boody", "margin-top" ) ).isEqualTo( "0px" );
 	}
+
+	@Test
+	void isDefault_handled_two_null_values() {
```
Comment (beatngu13) [inline src/test/java/de/retest/web/DefaultValuesProviderTest.java]: Doesn't the test tests more than "two null values"? It also checks if stuff like `auto`, `normal`, and `0px` is recognized as a default, besides the defaults provided in `defaults.yml`.

File: src/main/java/de/retest/web/DefaultValuesProvider.java [hunk 2]
```diff
@@ -74,8 +76,8 @@ public Serializable getDefaultValue( final IdentifyingAttributes comp, final Str
 
 	public boolean isDefault( final String tag, final String attribute, final String attributeValue ) {
 		final String defaultValue = getDefaultValue( tag, attribute );
-		if ( defaultValue != null ) {
-			return defaultValue.equalsIgnoreCase( attributeValue );
+		if ( StringUtils.equalsIgnoreCase( defaultValue, attributeValue ) ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/DefaultValuesProvider.java]: How about moving it behind the `null` check for `attributeValue` and then to:

```java
if ( attributeValue.equalsIgnoreCase( defaultValue ) ) { ... }
```

Comment (kema91) [reply src/main/java/de/retest/web/DefaultValuesProvider.java]: do you mean something like that: 


`if ( attributeValue == null || attributeValue.trim().isEmpty() ) {`
    `return true;`
`}`

`if ( attributeValue.equalsIgnoreCase( defaultValue ) {`
    ` return true;`
`}`

File: src/test/java/de/retest/web/DefaultValuesProviderTest.java (hunk 1 above)
Comment (kema91) [reply src/test/java/de/retest/web/DefaultValuesProviderTest.java]: I'll change it

Comment (beatngu13) [reply src/test/java/de/retest/web/DefaultValuesProviderTest.java]: I'll give you a call. 😉

File: src/main/java/de/retest/web/DefaultValuesProvider.java [hunk 3]
```diff
@@ -74,12 +74,12 @@ public Serializable getDefaultValue( final IdentifyingAttributes comp, final Str
 
 	public boolean isDefault( final String tag, final String attribute, final String attributeValue ) {
 		final String defaultValue = getDefaultValue( tag, attribute );
-		if ( defaultValue != null ) {
-			return defaultValue.equalsIgnoreCase( attributeValue );
-		}
 		if ( attributeValue == null || attributeValue.trim().isEmpty() ) {
 			return true;
 		}
+		if ( attributeValue.equalsIgnoreCase( defaultValue ) ) {
```
Comment (roesslerj) [inline src/main/java/de/retest/web/DefaultValuesProvider.java]: Damn, I overlooked that. We want to be able to say, that null is not the default value...

**Pull request merged**